### PR TITLE
Log diagnostics at the info level

### PR DESCRIPTION
**Pull request by djmitche**
This avoids spurious error-level output when running `agent flare` or
`agent diagnose`.

### Motivation

The error logs end up sent to DD where they can cause some confusion as to where that particular error came from.

### Additional Notes

This also applies a small refactor to log the error once, in the
`diagnose` runner, rather than in each diagnostic function.

### Describe how to test your changes

Run `agent diagnose` and see the level of the resulting logs.

### Checklist

- [x] A [release note](https://github.com/DataDog/datadog-agent/blob/main/docs/dev/contributing.md#reno) has been added or the `changelog/no-changelog` label has been applied.
- [x] The appropriate `team/..` label has been applied, if known.
- [x] If known, an appropriate milestone has been selected; otherwise the `Triage` milestone is set.

### CODE REVIEW

File: releasenotes/notes/diagnose-info-log-ae1b2adf9a60ae58.yaml
Comment (gh123man) [inline releasenotes/notes/diagnose-info-log-ae1b2adf9a60ae58.yaml:4]:
```suggestion
    The ``agent diagnose`` and ``agent flare`` commands no longer create error-level log messages when the diagnostics fail.
```

Comment (djmitche) [reply releasenotes/notes/diagnose-info-log-ae1b2adf9a60ae58.yaml:4]: This is one of those personal failings I've never been able to shake: awful typos in whatever textual thing I edited last before making a PR :(

**Review by ogaca-dd — APPROVED** [review]

File: pkg/util/ecs/diagnosis.go
Comment (olivielpeau) [inline pkg/util/ecs/diagnosis.go:61]: You probably meant instead:

```suggestion
		return fmt.Errorf("error while initializing ECS metadata V2 client: %w", err)
```

File: releasenotes/notes/diagnose-info-log-ae1b2adf9a60ae58.yaml
Comment (truthbk) [inline releasenotes/notes/diagnose-info-log-ae1b2adf9a60ae58.yaml:4]: I am not 100% sure, but we use pandoc to generate markdown from the RST, and we then use that on GH releases, the double back-ticks might cause some issues. Definitely something we can find out when the time comes.

File: pkg/diagnose/runner.go
Comment (truthbk) [inline pkg/diagnose/runner.go:46]: I'm fine with this, but wondering if we should maybe be a bit explicit in case the returned error isn't particularly well worded:
```
log.Infof("diagnosis error for %s: %v", name, err)
```

**Review by truthbk — APPROVED** [review]
Looks mostly ready, I made a small comment, but feel free to ignore if you disagree.

File: releasenotes/notes/diagnose-info-log-ae1b2adf9a60ae58.yaml
Comment (djmitche) [reply releasenotes/notes/diagnose-info-log-ae1b2adf9a60ae58.yaml:4]: I think ``..`` is [proper rST syntax](https://docutils.sourceforge.io/docs/user/rst/quickstart.html) -- it seems if pandoc is doing its job, it should work?  Let's find out :)

File: pkg/diagnose/runner.go
Comment (djmitche) [reply pkg/diagnose/runner.go:46]: Good point.  I'll make that change and merge.

Comment (djmitche) [reply pkg/diagnose/runner.go:46]: ..if it's green

**Comment by djmitche** [comment]
..and it's green!

**Comment by djmitche** [comment]
(but I'll wait for the remaining reviews)

File: pkg/util/ecs/diagnosis.go
Comment (olivielpeau) [reply pkg/util/ecs/diagnosis.go]: Thanks for addressing this, no need for a full approval from me

File: releasenotes/notes/diagnose-info-log-ae1b2adf9a60ae58.yaml
Comment (djmitche) [reply releasenotes/notes/diagnose-info-log-ae1b2adf9a60ae58.yaml:4]: (note: I saw some other double-backticks in other PRs, so I'm pretty sure this is OK)

**Review by ahmed-mez — APPROVED** [review]

**Review by nammn — APPROVED** [review]

**Pull request merged**
